pageserver/src/tenant/timeline/compaction.rs

```diff
@@ -1171,8 +1167,8 @@ impl Timeline {
                 // This ancestral layer only covers keys that belong to other shards.
                 // We include the full metadata in the log: if we had some critical bug that caused
                 // us to incorrectly drop layers, this would simplify manually debugging + reinstating those layers.
-                info!(%layer, old_metadata=?layer.metadata(),
-                    "dropping layer after shard split, contains no keys for this shard.",
+                debug!(%layer, old_metadata=?layer.metadata(),
```

Comment:
if we moved the print to later, why not remove this entirely?

Reply:
Just to have some debug-logging during the preparation loop, since every branch has debug logging.